contracts/epochs/FlowDKG.cdc

```diff
@@ -100,6 +90,230 @@ access(all) contract FlowDKG {
         }
     }
 
+    // Checks whether the ResultSubmission constructor arguments constitute a valid empty submission.
+    // A valid empty submission has all fields nil, and is used when the submittor locally failed the DKG.
+    // All non-empty submissions must have all non-nil fields.
+    access(all) view fun isValidNilSubmission(groupPubKey: String?, pubKeys: [String]?, idMapping: {String:Int}?): Bool {
+        // If any fields are nil, then this represents a empty submission and all fields must be nil
+        if groupPubKey == nil && pubKeys == nil && idMapping == nil {
+            return true
+        }
+        // Otherwise, all fields must be non-nil
+        return groupPubKey != nil && pubKeys != nil && idMapping != nil 
```

Comment:
I'm a little confused. The comments make it seem like this function is checking if any submission is valid, but the function name is `isValidNilSubmission`, which makes me think that this should only verify that the all-`nil` case is valid. Maybe it should be renamed to `isValidSubmission`? Also, we can simplify the conditional I think:
```
        return (groupPubKey == nil && pubKeys == nil && idMapping == nil)
        || (groupPubKey != nil && pubKeys != nil && idMapping != nil)
```

Reply:
The use-case for the function is in the constructor pre-condition, so we end up passing both empty and non-empty submissions to the function -- hence the weird naming. I agree it's confusing.

I don't love `isValidSubmission`, because not all the validity checks for a submission are contained in this function (you could pass `isValidSubmission` with an invalid submission). I renamed it to refer to the invariant it is enforcing in https://github.com/onflow/flow-core-contracts/pull/441/commits/07090b8ff8a8fb8bd0dc97f350b2a7c22c22e5a1.